server/src/domain/media/media.service.ts

```diff
@@ -65,22 +66,27 @@ export class MediaService {
     });
 
     for await (const assets of assetPagination) {
+      const jobs: JobItem[] = [];
+
       for (const asset of assets) {
         if (!asset.resizePath || force) {
-          await this.jobRepository.queue({ name: JobName.GENERATE_JPEG_THUMBNAIL, data: { id: asset.id } });
+          jobs.push({ name: JobName.GENERATE_JPEG_THUMBNAIL, data: { id: asset.id } });
           continue;
         }
         if (!asset.webpPath) {
-          await this.jobRepository.queue({ name: JobName.GENERATE_WEBP_THUMBNAIL, data: { id: asset.id } });
+          jobs.push({ name: JobName.GENERATE_WEBP_THUMBNAIL, data: { id: asset.id } });
         }
         if (!asset.thumbhash) {
-          await this.jobRepository.queue({ name: JobName.GENERATE_THUMBHASH_THUMBNAIL, data: { id: asset.id } });
+          jobs.push({ name: JobName.GENERATE_THUMBHASH_THUMBNAIL, data: { id: asset.id } });
         }
       }
+
+      await this.jobRepository.queueAll(jobs);
     }
 
     const people = force ? await this.personRepository.getAll() : await this.personRepository.getAllWithoutThumbnail();
 
+    const idsToGenerateThumbnail = new Set<string>();
```

Comment:
Why do you use a set of strings here and not just an array of `{name, data}` objects like in the other places?

Reply:
Good call! I used this way in case we could be receiving duplicate ids, but I can see that's not possible in any of these two places, so building the array of jobs directly will be simpler.